pkg/services/sqlstore/user.go

```diff
@@ -620,6 +621,21 @@ func (ss *SQLStore) SearchUsers(ctx context.Context, query *models.SearchUsersQu
 			whereParams = append(whereParams, query.OrgId)
 		}
 
+		// user only sees the users in which it has user read access too
+		if ss.Cfg.IsFeatureToggleEnabled(featuremgmt.FlagAccesscontrol) && query.SignedInUser != nil {
+			acFilter, err := ac.Filter(
+				query.SignedInUser,
+				"u.id",
+				"users:id:",
+				ac.ActionOrgUsersRead,
```

Comment:
It seems that the search user endpoint is using this and is protected with `ac.EvalPermission(ac.ActionUsersRead, ac.ScopeGlobalUsersAll))`. Did you try the following?
```suggestion
				"global.users:id:",
				ac.ActionUsersRead,
```

Reply:
ahhh i did not do it with `global` 😶‍🌫️ . then it probably works